tests_new/integration_tests/core/environment/test_environment.py

```diff
@@ -52,3 +60,60 @@ def test_persistent_env_update(client1, persistent_env1):
     check_stack_in_progress(client1, env_uri, stack_uri)
     stack = check_stack_ready(client1, env_uri, stack_uri)
     assert_that(stack.status).is_equal_to('UPDATE_COMPLETE')
+
+
+def test_invite_group_on_env_no_org(client1, session_env2, group3):
+    assert_that(invite_group_on_env).raises(GqlError).when_called_with(
+        client1, session_env2.environmentUri, group3, ['CREATE_DATASET']
+    ).contains(group3, 'is not a member of the organization')
+
+
+def test_invite_group_on_env_unauthorized(client2, session_env2, group2):
+    assert_that(invite_group_on_env).raises(GqlError).when_called_with(
+        client2, session_env2.environmentUri, group2, ['CREATE_DATASET']
+    ).contains('UnauthorizedOperation', 'INVITE_ENVIRONMENT_GROUP', session_env2.environmentUri)
+
+
+def test_invite_group_on_env(client1, client2, session_env2, group2):
+    env_uri = session_env2.environmentUri
+    assert_that(list_environments(client2).nodes).extracting('environmentUri').contains(env_uri)
+    # assert that client2 can get the environment
+    assert_that(get_environment(client2, env_uri)).contains_entry(userRoleInEnvironment='Invited')
+
+
+def test_invite_remove_group_on_env(client1, client3, session_env2, group3):
+    env_uri = session_env2.environmentUri
+    try:
+        assert_that(list_environments(client3).nodes).extracting('environmentUri').does_not_contain(env_uri)
+        assert_that(invite_group_on_env(client1, env_uri, group3, ['CREATE_DATASET'])).contains_entry(
+            environmentUri=env_uri
+        )
+        # assert that client3 can get the environment
+        assert_that(get_environment(client3, env_uri)).contains_entry(userRoleInEnvironment='Invited')
+    finally:
+        assert_that(remove_group_from_env(client1, env_uri, group3)).contains_entry(environmentUri=env_uri)
+        assert_that(get_environment).raises(GqlError).when_called_with(client3, env_uri).contains(
+            'UnauthorizedOperation', 'GET_ENVIRONMENT', env_uri
+        )
+
+
+def test_add_remove_consumption_role(client1, session_env2, group1):
+    env_uri = session_env2.environmentUri
+    consumption_role = None
+    try:
+        consumption_role = add_consumption_role(
+            client1, env_uri, group1, 'TestConsumptionRole', f'arn:aws:iam::{session_env2.AwsAccountId}:role/Admin'
+        )
```

Comment:
Are we sure `arn:aws:iam::{session_env2.AwsAccountId}:role/Admin` will always be a valid role ARN that we can use for `add_consumption_role`? Is there any need to validate or create IAM role if not exists before runnign add_consumption_role (++ clean up)?

Reply:
I think Admin is always there in case I am wrong we need to do what you suggest.
Would you mind if we leave that for later and in case Admin is not there?